Tko jan24

tkoukkari opened the pull request:
* Feature: Adds new table keywords for handling column headers:
  * GetColHeaderCount
  * GetColHeader
  * VerifyColHeader
* Feature: Support Robot Framework 7 data model in TypeSecret
* Fix: Handle booleans given in string format in OpenBrowser prefs (i.e… "True")
* Dependencies: 
  * Allow Robot Framework 7
  * Bumped lowest allowed Pillow version 
* Updated workflow actions to newer versions

tkoukkari committed:
tests: updating workflow actions to later versions

tkoukkari committed:
docs: Added note and examples to GetText for using one char separator when using between argument

tkoukkari committed:
feat: added new table keywords for handling columns:
- GetColumnCount
- GetColumn
- VerifyColumn

tkoukkari committed:
deps: allow robot framework 7

tkoukkari committed:
Revert "deps: allow robot framework 7"

This reverts commit d01e875beaa1e82353d198fd897874b4dbdd7390.

tkoukkari committed:
tests: added longer timeout for first suite
...in case env setup takes longer in github

tkoukkari committed:
chore: allow robot framework 7
+ fix typing warnings from new rfw version

tkoukkari committed:
feat: support for rfw7 data model (TypeSecret)

tkoukkari committed:
fix: handle booleans given in string format in OpenBrowser prefs (i.e. "True")

tkoukkari committed:
deps: bump lowest allow Pillow version, includes security fixes

tkoukkari committed:
test: additional verification to openbrowser prefs verification case

turunenm commented on QWeb/keywords/cookies.py:
```diff
@@ -64,7 +64,7 @@ def list_cookies() -> list[dict[str, Any]]:
     """
     cookies_list = cookies.get_cookies()
-    logger.info(cookies_list)
+    logger.info(f"{cookies_list}")
```
This is off topic from the PR but I've been wondering what is the rationale to info logging the cookies automatically.
I have some scripts where I'm looking for cookie values to update on a rapid pace and it clogs up the report.

Maybe add argument `log_cookies=True` to provide control over it for the user?

tkoukkari replied:
Good suggestion, will be handled in own pr / part of one release after this one.

turunenm commented on QWeb/keywords/table.py:
```diff
@@ -295,3 +296,199 @@ def get_table_row(
+
+
+@keyword(tags=("Tables", "Getters"))
+def get_column(index: Optional[int] = None) -> Union[str, List[str]]:
```
I would change this to `get_header_column` or `get_table_header_column`

Compare the behavior to `get_table_row` which returns the first row which contains the sought text.
This kw only looks for the first row and it's column count which differs from the existing row kw

tkoukkari replied:
Renamed as:
- GetColumnCount -> GetColHeaderCount
- GetColumn -> GetColHeader
- VerifyColumn -> VerifyColHeader

turunenm commented on QWeb/keywords/table.py:
```diff
@@ -295,3 +296,199 @@ def get_table_row(
+
+
+@keyword(tags=("Tables", "Getters"))
+def get_column_count() -> int:
```
I would change this to `get_header_column_count` or `get_table_header_column_count`

Rows can have differing amounts of columns which are close impossible to account for in a sane manner.
But even so the kw name should describe as accurately as possible what it actually does.

tkoukkari replied:
Renamed in latest commit

tkoukkari replied:
Renamed:
- GetColumnCount -> GetColHeaderCount
- GetColumn -> GetColHeader
- VerifyColumn -> VerifyColHeader

turunenm commented on QWeb/keywords/table.py:
```diff
@@ -295,3 +296,199 @@ def get_table_row(
+    If not and 'title' attirbute exists, then title is considered as expected value.
+    If neither aria-label or title exists, then column text is considered as expected value.
+    This is done due to fact that in many web applications header element text contains other texts
+    than what is visible to end users.
```
Doc and behavior do not match.

Looking at `_get_column_header_texts` -> title > aria-label > text

tkoukkari replied:
Thanks, did not notice this. Fixed in latest commit and also the typo in help text fixed

turunenm commented on QWeb/keywords/table.py:
```diff
@@ -295,3 +296,199 @@ def get_table_row(
+
+
+@keyword(tags=("Tables", "Verification"))
+def verify_column(expected: str,
```
I would change this to `verify_header_column`

The kw only looks for the first rows column texts and the name should reflect it

tkoukkari replied:
renamed

turunenm commented on QWeb/internal/secrets.py:
```diff
@@ -101,6 +109,34 @@ def _filtered_start_keyword(keyword: Keyword) -> None:
             LOGGER._other_loggers[0].log_message = lambda x: None
 
 
+def _filtered_start_library_keyword(data: Keyword, implementation: Keyword, result: Keyword):
```
Add docstring.
Specify that this is function for RFW version 7+

tkoukkari replied:
added and also end_library_keyword docstring modified

turunenm commented on QWeb/internal/secrets.py:
```diff
@@ -130,6 +166,37 @@ def _filtered_end_keyword(keyword: Keyword) -> None:
+
+    This function removes secret data from "end keyword"
+    logs and returns previous log level.
+    """
```
Specify that this function is for RFW version 7+

tkoukkari replied:
done

turunenm reviewed (commented):
IMO column kw names should be changed to reflect their actual behavior to avoid confusion.

tkoukkari committed:
refactor: renamed column keywords
- GetColumnCount -> GetColHeaderCount
- GetColumn -> GetColHeader
- VerifyColumn -> VerifyColHeader

These make it more obvious what kw does without reading docs.

tkoukkari committed:
docs: fixed col header search order as defined in docs + typos

tkoukkari committed:
docs: modified docstrings for rw7 start_/end_library_keyword patches

turunenm reviewed (approved):
lgtm
